GalaxyZooTouchTable/GalaxyZooTouchTable/Services/ILocalDBService.cs

```diff
@@ -9,9 +9,9 @@ public interface ILocalDBService
     {
         TableSubject GetLocalSubject(string id);
 
-        List<TableSubject> GetLocalSubjects(SpaceNavigation currentLocation);
+        List<TableSubject> GetLocalSubjects(SpaceNavigation currentLocation, bool returnIfRetired = false);
```

Comment:
Is there a circumstance under which would we want returnIfRetired to be false, or do we always want this to be the case?

Reply:
There's a use case for both circumstances. I'm mainly trying to avoid being caught in an infinite loop where the local database is constantly being queried for subjects but can't find any because they're all retired. This is a definite possibility with the added [classification limits](https://github.com/zooniverse/galaxy-zoo-touch-table/pull/81/files/e204073f77317689fa3b461bae38be53b383beca#diff-7555746014045015b676c065b11e05dfR22) to each many of the database queries. 

What would happen if _every_ subject on the table already has 25 classifications? It's rare that would happen, but a possibility. I don't want the table to crash or stop navigating because there are no subjects to navigate. In this case, a subject should still be loaded, even if it has retired. 

Ex: We move west by running [this method](https://github.com/zooniverse/galaxy-zoo-touch-table/pull/81/files/e204073f77317689fa3b461bae38be53b383beca#diff-03461c6521d90ee0752bff2604107b44R202). The app first checks if any galaxies are one cutout image west. If [none exist there](https://github.com/zooniverse/galaxy-zoo-touch-table/pull/81/files/e204073f77317689fa3b461bae38be53b383beca#diff-03461c6521d90ee0752bff2604107b44R253), the app tries to find the [next western-most galaxy](https://github.com/zooniverse/galaxy-zoo-touch-table/pull/81/files/e204073f77317689fa3b461bae38be53b383beca#diff-7555746014045015b676c065b11e05dfR162). That process looks like this: 

- Find the galaxy with the next highest right ascension (RA) value, ascending
- If that doesn't exist, find the galaxy with the lowest RA value, as we're gone around in a circle in space where 360 degrees RA goes back to zero degrees
- If you still can't find a galaxy because they're all retired, just load any random galaxy with the least amount of classifications

The third point is where we want to return subjects from the local database, even if they're retired.